### PR TITLE
feat: create a settings modal for the explorer

### LINKED ISSUE #900: feat: settings page
Currently there are a few options in the explorer that can be customized:

- language
- currency
- chart visibility
- theme

These options should move to a separate settings page where users can set these values, instead of having the options in the header of the explorer. This will open up the possibility to add more setting options in the near future

### PR DESCRIPTION
**Pull request by brenopolanski**
## Summary

Closes #900.

Screenshot from 2020-05-15 13-47-27

## Checklist

- [ ] Documentation _(if necessary)_
- [x] Tests _(if necessary)_
- [x] Ready to be merged

### CODE REVIEW

**Comment by ItsANameToo** [comment]
a few things:

- buttons should be like this: 
image
currently the save button looks like it's disabled instead of standing out
- can you change "translations" to "language"
- `Customize your wallet to suit your needs` should be `Customize the Explorer to suit your needs` 
- for the languages, it would be better if we show the actual language name instead of the abbreviation. So en-gb would become "English (UK)" and nl-NL would be "Nederlands" (in native language).
- currently tests are failing so please look into those
- Personally I don't see the need for the guiding texts on desktop screens, especially as they are hidden on mobile, so I'll ask for more input on that on slack

otherwise looks great and works as intended 👌

**Comment by brenopolanski** [comment]
- [x] Update save button (UI);
- [x] Change `translations` to `language`;
- [x] `Customize your wallet to suit your needs` should be `Customize the Explorer to suit your needs`;
- [x] Use language name instead of the abbreviation;
- [x] Fix tests;
- [x] Update text guides.

File: src/components/modals/SettingsModal.vue [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    this.$store.dispatch("ui/setLanguage", language);
+    this.$i18n.locale = language;
+
+    this.$store.dispatch("ui/setLocale", language);
+    moment.locale(language);
+  }
+
+  private toggleTheme(enabled: boolean) {
+    this.nightMode = enabled;
+  }
```
Comment (ItsANameToo) [inline src/components/modals/SettingsModal.vue]:
```suggestion
  private toggleTheme() {
    this.nightMode = !this.nightMode;
  }
```

similar for `toggleChart`, as it's not really toggling if you make it set a specific value

File: src/assets/css/_buttons.css [hunk 2]
```diff
@@ -40,6 +40,13 @@ button:disabled {
   @apply .bg-blue .text-white .hover-button-shadow-definition .cursor-pointer;
 }
 
+.action-button {
+  @apply .base-button .bg-blue .text-white .hover-button-shadow-definition .py-2 .px-4;
```
Comment (ItsANameToo) [inline src/assets/css/_buttons.css]: move `.hover-button-shadow-definition` to `.action-button:enabled:hover`

File: src/assets/css/_buttons.css [hunk 3]
```diff
@@ -40,6 +40,13 @@ button:disabled {
...
+}
+
+.action-button:enabled:hover {
+  @apply .opacity-75 .cursor-pointer;
```
Comment (ItsANameToo) [inline src/assets/css/_buttons.css]: remove opacity

File: src/config/index.ts [hunk 4]
```diff
@@ -1,6 +1,15 @@
...
   defaultLocale: "en-GB",
   enabledLocales: ["en-GB", "en-US", "it-IT", "fr-FR", "nl-NL", "pl-PL", "pt-BR"],
+  languages: {
+    "en-GB": "English (UK)",
+    "en-US": "English (US)",
+    "it-IT": "Italian",
+    "fr-FR": "French",
+    "nl-NL": "Nederlands",
+    "pl-PL": "Polish",
+    "pt-BR": "Portuguese (BR)",
```
Comment (ItsANameToo) [inline src/config/index.ts]:
```suggestion
  languages: {
    "en-GB": "English (UK)",
    "en-US": "English (US)",
    "it-IT": "Italiano",
    "fr-FR": "Français",
    "nl-NL": "Nederlands",
    "pl-PL": "Polski",
    "pt-BR": "Português (BR)",
```

File: src/components/modals/SettingsModal.vue [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+          <ListDividedItem
+            :label="$t('MODAL_SETTINGS.PRICE_CHART.LABEL')"
+            :label-description="$t('MODAL_SETTINGS.PRICE_CHART.DESCRIPTION')"
+            item-label-description-class="hidden md:block"
+          >
+            <ButtonSwitch :is-active="chartMode" class="SettingsModal__toggle__priceChart mt-2" @change="toggleChart" />
+          </ListDividedItem>
+          <ListDividedItem
+            :label="$t('MODAL_SETTINGS.LANGUAGE.LABEL')"
+            :label-description="$t('MODAL_SETTINGS.LANGUAGE.DESCRIPTION')"
```
Comment (ItsANameToo) [inline src/components/modals/SettingsModal.vue]: can you remove the descriptions? they aren't really needed for these simple settings

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

File: src/components/modals/SettingsModal.vue (hunk 5 above)
Comment (brenopolanski) [reply src/components/modals/SettingsModal.vue]: @ItsANameToo just to confirm, remove all descriptions?

Comment (ItsANameToo) [reply src/components/modals/SettingsModal.vue]: correct

**Comment by ItsANameToo** [comment]
Almost there @brenopolanski , final changes:

- when you are on devnet, there are no currencies defined so it should hide that option.
- similar as above for the price chart, if it's not an enabled option for the network it should not be shown in the settings

**Comment by brenopolanski** [comment]
> Almost there @brenopolanski , final changes:
> 
> * when you are on devnet, there are no currencies defined so it should hide that option.
> * similar as above for the price chart, if it's not an enabled option for the network it should not be shown in the settings

Done @ItsANameToo

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**